### PR TITLE
Task7

### PR DESCRIPTION
**Pull request by uladzimir-yeudakimovich**
1 - authorization-service is added to the repo, has correct basicAuthorizer lambda and correct serverless.yaml file
3 - import-service serverless.yaml file has authorizer configuration for the importProductsFile lambda. Request to the importProductsFile lambda should work only with correct authorization_token being decoded and checked by basicAuthorizer lambda. Response should be in 403 HTTP status if access is denied for this user (invalid authorization_token) and in 401 HTTP status if Authorization header is not provided.
5 - update client application to send Authorization: Basic authorization_token header on import. Client should get authorization_token value from browser localStorage

Front-end link: https://d2lsb4xgmff0fw.cloudfront.net/admin/products
authorization_token=dWxhZHppbWlyeWV1ZGFraW1vdmljaDpURVNUX1BBU1NXT1JE
products.scv file in the root of import-servise

+1 - Client application should display alerts for the responses in 401 and 403 HTTP statuses. This behavior should be added to the nodejs-aws-fe-main/src/index.tsx file

https://github.com/uladzimir-yeudakimovich/nodejs-aws-fe/pull/5

Cognito Authorizer link: https://macbook-shop.auth.eu-west-1.amazoncognito.com/login?client_id=92sc5jcq88eui6rck98dkgbno&response_type=token&scope=aws.cognito.signin.user.admin+email+openid+phone+profile&redirect_uri=http://localhost:3000/

+ add cognito auth for getProductsList

### CODE REVIEW

File: authorization-service/handlers/requestAuthorizer.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+export const requestAuthorizer = async (event, ctx, cb) => {
```
Comment (Shutya) [inline authorization-service/handlers/requestAuthorizer.js:1]: What is the main purpose of requestAuthorizer ? As I can see, you don't use it anywhere.

File: product-service/serverless.ts [hunk 2]
```diff
@@ -76,7 +116,13 @@ const serverlessConfiguration: Serverless = {
           http: {
             method: 'get',
             path: 'products',
-            cors: true
+            cors: true,
+            authorizer: {
+              name: 'cognitoAuthorizer',
+              type: 'COGNITO_USER_POOLS',
+              arn: 'arn:aws:cognito-idp:eu-west-1:714652663732:userpool/eu-west-1_zTPf140Wi',
```
Comment (Shutya) [inline product-service/serverless.ts:123]: Seems like you know it, but better to use AWS::Region and AWS::AccountId here as you've done in importProductsFile.

**Comment by svetlanagorchanina** [comment]
Good job!
Some minor notes:
- the following request will have status 200, but it's not "Basic" authentication
```
fetch("https://xu0z0hbqm1.execute-api.eu-west-1.amazonaws.com/dev/import?name=test.csv", {
  "headers": {
    "authorization": "Test dWxhZHppbWlyLXlldWRha2ltb3ZpY2g6VEVTVF9QQVNTV09SRA==",
  },
});
```

File: authorization-service/handlers/requestAuthorizer.js (hunk 1 above)
Comment (uladzimir-yeudakimovich) [reply authorization-service/handlers/requestAuthorizer.js:1]: Main purpose - training )))

File: product-service/serverless.ts (hunk 2 above)
Comment (uladzimir-yeudakimovich) [reply product-service/serverless.ts:123]: Thanks, forgot to replace here

**Pull request merged**